tripal_chado/src/Plugin/Field/FieldType/ChadoContactByRoleTypeDefault.php

```diff
@@ -0,0 +1,302 @@
+      'drupal_store' => FALSE,
+      'path' => $table_alias . '.' . $linker_fkey_column . '>' . $object_table . '.' . $object_pkey_col . ';description',
+      'table_alias_mapping' => [$table_alias => $linker_table],
+      'as' => 'contact_description',
+    ]);
+
+    // The type of contact
+    $properties[] = new ChadoVarCharStoragePropertyType($entity_type_id, self::$id, 'contact_type', $terms['contact_type'], $max_lengths['contact_type'], [
+      'action' => 'read_value',
+      'drupal_store' => FALSE,
+      'path' => $table_alias . '.' . $linker_fkey_column . '>' . $object_table . '.' . $object_pkey_col
+        . ';' . $object_table . '.type_id>cvterm.cvterm_id;name',
+      'table_alias_mapping' => [$table_alias => $linker_table],
+      'as' => 'contact_type',
+    ]);
+
+    return $properties;
+  }
+
+  /**
+   * We need to set the type_id property value to match the cvterm_id.
+   *
+   * To do this we'll override the tripalValuesTemplate() and give the
+   * `type_id` property a default value.
+   *
+   * {@inheritDoc}
+   * @see \Drupal\tripal\TripalField\TripalFieldItemBase::tripalValuesTemplate()
+   */
+  public function tripalValuesTemplate($field_definition, $default_value = NULL) {
+    $prop_values = parent::tripalValuesTemplate($field_definition, $default_value);
+
+    $settings = $field_definition->getSettings();
+
+    $termIdSpace = $settings['termIdSpace'];
+    $termAccession = $settings['termAccession'];
+
+    /** @var \Drupal\tripal\TripalVocabTerms\PluginManagers\TripalIdSpaceManager $idSpace_manager **/
+    /** @var \Drupal\tripal\TripalVocabTerms\TripalIdSpaceBase $idSpace **/
+    /** @var \Drupal\tripal\TripalVocabTerms\TripalTerm $term **/
+    $idSpace_manager = \Drupal::service('tripal.collection_plugin_manager.idspace');
+    $idSpace = $idSpace_manager->loadCollection($termIdSpace);
+    $term = $idSpace->getTerm($termAccession);
+
+    foreach ($prop_values as $index => $prop_value) {
+      if ($prop_value->getKey() == 'linker_type_id') {
+        $prop_values[$index]->setValue($term->getInternalId());
+      }
+    }
+
+    return $prop_values;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\tripal_chado\TripalField\ChadoFieldItemBase::isCompatible()
+   */
+  public function isCompatible(TripalEntityType $entity_type) : bool {
+
+    // Get the base table for the content type.
+    $has_linker = FALSE;
+    $base_table = $entity_type->getThirdPartySetting('tripal', 'chado_base_table');
+    // Get the list of tables linking the base table and our object table.
+    $linker_tables = $this->getLinkerTables(self::$object_table, $base_table);
+    if (!empty($linker_tables)) {
+      $has_linker = TRUE;
+    }
+
+    // Ensure that the linker table has a type_id.
+    $has_type_id = FALSE;
+    foreach ($linker_tables as $item) {
+
+      [$table_name, $contact_fkey] = $item;
+
+      // Check there is a type_id field.
+      $schemaObj = \Drupal::service('tripal_chado.database')->schema();
```

Comment:
```suggestion
    $schemaObj = \Drupal::service('tripal_chado.database')->schema();
    foreach ($linker_tables as $item) {

      [$table_name, $contact_fkey] = $item;

      // Check there is a type_id field.
```
This is absolutely not likely to be significant, but for performance we would only need to retrieve $schemaObj once